src/backend/distributed/executor/executor.c

```diff
@@ -1063,17 +1088,14 @@ RunDistributedExecution(DistributedExecution *execution)
 		 */
 		UnclaimAllSessionConnections(execution->sessionList);
 
-		/* make sure the epoll file descriptor is always closed */
-		if (waitEventSet != NULL)
-		{
-			FreeWaitEventSet(waitEventSet);
-		}
-
+		FreeWaitEventSet(waitEventSet);
 		pfree(events);
 
 		PG_RE_THROW();
 	}
 	PG_END_TRY();
+
+	FreeWaitEventSet(waitEventSet);
```

Comment:
this could maybe just be after `pfree(events);`

Reply:
done